From 25d2af4e3e0d08193395df2cb22eb474b0e1d5ab Mon Sep 17 00:00:00 2001
From: Palmer Dabbelt <palmer.dabbelt@eecs.berkeley.edu>
Date: Fri, 20 Dec 2013 18:57:16 -0800
Subject: [PATCH] Add Configfile for pconfigure

pconfigure is a build system that I wrote.  I've been maintaining a
Chisel fork for a while that builds with pconfigure.  Over break I'm
trying to get some of my forks merged, so I figured this would be a
good way to start.

This shouldn't cause anyone trouble because the sbt stuff is
completely untouched -- in other words if you don't want to use
pconfigure (ie, everyone else :)) then just ignore it!
---
 .gitignore    |  7 +++++++
 Configfile    | 13 +++++++++++++
 src/chisel.pc | 11 +++++++++++
 3 files changed, 31 insertions(+)
 create mode 100644 Configfile
 create mode 100644 src/chisel.pc

diff --git a/.gitignore b/.gitignore
index a31c0b7..988d242 100644
--- a/.gitignore
+++ b/.gitignore
@@ -10,3 +10,10 @@ target/
 *.toc
 *.bak
 *.new
+
+# Files auto-generated by pconfigure
+/Configfile.local
+/Makefile
+/lib/
+/obj/
+/check/
diff --git a/Configfile b/Configfile
new file mode 100644
index 0000000..8a2e055
--- /dev/null
+++ b/Configfile
@@ -0,0 +1,13 @@
+LANGUAGES += scala
+LANGUAGES += pkgconfig
+LANGUAGES += h
+
+LIBRARIES += libchisel.jar
+SOURCES   += main/scala/Cpp.scala
+
+LIBRARIES += pkgconfig/chisel.pc
+SOURCES   += chisel.pc
+
+HEADERS   += emulator.h
+SRCDIR     = csrc
+SOURCES   += emulator.h
diff --git a/src/chisel.pc b/src/chisel.pc
new file mode 100644
index 0000000..af2c664
--- /dev/null
+++ b/src/chisel.pc
@@ -0,0 +1,11 @@
+prefix=@@pconfigure_prefix@@
+exec_prefix=@@pconfigure_prefix@@
+libdir=${prefix}/@@pconfigure_libdir@@
+includedir=${prefix}/@@pconfigure_hdrdir@@
+
+Name: chisel
+Description: The Chisel wiring language
+Version: 2.0.0pc
+CFlags: -I${includedir}
+Libs: -L${libdir} -lchisel
+URL: http://github.com/ucb-bar/chisel
-- 
1.8.3.2

